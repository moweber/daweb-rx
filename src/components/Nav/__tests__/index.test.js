import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Navbar from "..";

const mockCSection = "about";
const mockSetCSection = jest.fn();

afterEach(cleanup);

describe("Nav component renders", () => {
  it("renders", () => {
    render(<Navbar cSection={mockCSection} setCSection={mockSetCSection} />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Navbar cSection={mockCSection} setCSection={mockSetCSection} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Developer's name present", () => {
  it("inserts text into Brand", () => {
    const { getByTestId } = render(
      <Navbar cSection={mockCSection} setCSection={mockSetCSection} />
    );

    expect(getByTestId("developer")).toHaveTextContent("Matt Weber");
  });
});

describe("Links are visible", () => {
  it("inserts text into the links", () => {
    const { getByTestId } = render(
      <Navbar cSection={mockCSection} setCSection={mockSetCSection} />
    );

    expect(getByTestId("about")).toHaveTextContent("About");
    expect(getByTestId("portfolio")).toHaveTextContent("Portfolio");
    expect(getByTestId("contact")).toHaveTextContent("Contact");
    expect(getByTestId("resume")).toHaveTextContent("Resume");
  });
});
