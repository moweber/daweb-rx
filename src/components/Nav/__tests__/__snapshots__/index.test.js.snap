// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="fluid-container mNavbar"
  >
    <ul>
      <li
        class="mBrand"
      >
        <button
          data-testid="developer"
        >
          Matt Weber
        </button>
      </li>
      <li>
        <button
          class="navActive"
          data-testid="about"
        >
          About Me
        </button>
      </li>
      <li>
        <button
          class="false"
          data-testid="portfolio"
        >
          Portfolio
        </button>
      </li>
      <li>
        <button
          class="false"
          data-testid="contact"
        >
          Contact
        </button>
      </li>
      <li>
        <button
          class="false"
          data-testid="resume"
        >
          Resume
        </button>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
